File: 1859-sorting-the-sentence/1859-sorting-the-sentence.cpp
class Solution {
public:
    string sortSentence(string s) {
        unordered_map<string,int> mp;
          stringstream ss(s);
          string word;
          string st="";
          vector<string> v;
      
          while(ss >> word){
             int dig=word[word.length()-1]-'0';
             
             string w=word.substr(0,word.length()-1);
             mp[w]=dig;
            v.push_back(w);
          }
         
          sort(v.begin(),v.end(),[&](string a,string b){
            return mp[a] < mp[b];
          });
          for(int i=0;i<v.size();i++){
            st+=v[i]+" ";
          }
            st.pop_back();
            return st;
    }
};

File: 1859-sorting-the-sentence/1859-sorting-the-sentence.py
class Solution:
    def sortSentence(self, s: str) -> str:
        di={}
        for word in s.split():
            w=int(word[-1])
            l=word[:-1]
            di[w]=l
        return " ".join(di[i] for i in range(1,len(di)+1))
